frontend/test/metabase/scenarios/question/nested.cy.spec.js

```diff
@@ -423,6 +424,49 @@ describe("scenarios > question > nested", () => {
         });
     }
   });
+
+  ["count", "average"].forEach(test => {
+    it.skip(`${test.toUpperCase()}:\n should be able to use aggregation functions on saved native question (metabase#15397)`, () => {
+      cy.server();
+      cy.route("POST", "/api/dataset").as("dataset");
+
+      cy.createNativeQuestion({
+        name: "15397",
+        native: {
+          query:
+            "select count(*), orders.product_id from orders group by orders.product_id;",
+        },
+      });
+      cy.visit("/question/new");
+      cy.findByText("Simple question").click();
+      cy.findByText("Saved Questions").click();
+      cy.findByText("15397").click();
+      cy.wait("@dataset");
+      cy.findAllByText("Summarize")
+        .first()
+        .click();
+      if (test === "count") {
+        cy.findByText("Group by")
+          .parent()
+          .findByText("COUNT(*)")
+          .click();
+      } else {
+        sidebar()
+          .findByText("Count")
+          .should("be.visible")
+          .click();
+        cy.findByText("Average of ...").click();
+        popover()
+          .findByText("COUNT(*)")
+          .click();
+      }
+
+      cy.wait("@dataset").then(xhr => {
+        expect(xhr.response.body.error).not.to.exist;
+      });
+      cy.get(".bar").should("have.length.of.at.least", 20);
```

Comment:
The "average" forEach doesn't do a "Group by", so it will not return a bar.

Reply:
True.
I'll add another step to it so that both queries result in bar chart visualization.

Thank you.

Reply:
OK, fixed it by rearranging this test a little bit.
[THIS](https://github.com/metabase/metabase/pull/15406/files#diff-21278080a24869071372372d464993b5393bd276d1c08e3471f829a5af0fb311R448-R458) is the only part that is specific to "average" only. While [this](https://github.com/metabase/metabase/pull/15406/files#diff-21278080a24869071372372d464993b5393bd276d1c08e3471f829a5af0fb311R459-R462) is common for both cases.
Now we "group by" for both scenarios and will end up with bar visualizations.